src/Sulu/Bundle/AdminBundle/Resources/js/containers/Form/fields/ResourceLocator.js

```diff
@@ -19,7 +21,11 @@ class ResourceLocator extends React.Component<FieldTypeProps<?string>> {
     constructor(props: FieldTypeProps<?string>) {
         super(props);
 
-        const {dataPath, onChange, fieldTypeOptions, formInspector, value} = this.props;
+        this.state = {
```

Comment:
We don't use react state, because it is async and causes some issues... Use a MobX `@observable` variable as for the `mode` above. Then MobX will take care of rerendering the component, when you assign a new value to that variable.